packages/shared/src/hooks/useRegistration.ts

```diff
@@ -54,11 +54,23 @@ const useRegistration = ({
   const { displayToast } = useToastNotification();
   const { trackingId, referral, referralOrigin } = useContext(AuthContext);
   const timezone = getUserDefaultTimezone();
-  const { data: registration, isLoading: isQueryLoading } = useQuery(
-    key,
-    () => initializeKratosFlow(AuthFlow.Registration),
-    { refetchOnWindowFocus: false },
-  );
+  const {
+    data: registration,
+    isLoading: isQueryLoading,
+    error,
+  } = useQuery(key, () => initializeKratosFlow(AuthFlow.Registration), {
+    refetchOnWindowFocus: false,
+  });
+
+  if (error) {
+    trackEvent({
+      event_name: AuthEventNames.RegistrationInitialisationError,
+      extra: JSON.stringify({
+        error,
+        origin: 'initialize registration flow',
+      }),
+    });
+  }
```

Comment:
You need `useEffect` for this or try catch inside `queryFn` above right? Like this it will fire multiple times if parent hook/component/context re-renders.

Reply:
Yeah good point, might be an edge case where it re-renders after throwing the error.
Let me move it.

Reply:
@capJavert Please check of modification suits the need : )